src/containers/Tag/Tag.tsx

```diff
@@ -1,58 +1,159 @@
+        id
+        label
+        isActive
+        isReserved
+        description
+        language {
+          id
+        }
+      }
+      errors {
+        key
+        message
+      }
+    }
+  }
+`;
+
 export const Tag: React.SFC<TagProps> = (props) => {
-  const tagList = useSelector((state: AppState) => {
-    return state.tag.tags;
+  const tagId = props.match.params.id ? props.match.params.id : false;
+  const { loading, error, data } = useQuery(GET_TAG, {
+    variables: { id: tagId },
+    skip: !tagId,
   });
+  const [updateTag] = useMutation(UPDATE_TAG);
+  const languages = useQuery(GET_LANGUAGES);
 
-  const dispatch = useDispatch();
-  const onTagAdd = (tag: tagTypes.Tag) => {
-    dispatch(tagActions.addTag(tag));
-  };
+  const [label, setLabel] = useState('');
+  const [description, setDescription] = useState('');
+  const [isActive, setIsActive] = useState(false);
+  const [isReserved, setIsReserved] = useState(false);
+  const [languageId, setLanguageId] = useState(1);
+  const [parentId, setParentId] = useState('');
+  const [formSubmitted, setFormSubmitted] = useState(false);
 
-  const onTagEdit = (tag: tagTypes.Tag) => {
-    dispatch(tagActions.editTag(tag));
-  };
+  const [createTag] = useMutation(CREATE_TAG, {
+    update(cache, { data: { createTag } }) {
+      const tags: any = cache.readQuery({ query: GET_TAGS });
 
-  const tagId = props.match ? props.match.params.id : null;
-  const tag = tagId ? tagList.find((tag) => tag.id === Number(tagId)) : null;
+      cache.writeQuery({
+        query: GET_TAGS,
+        data: { tags: tags.tags.concat(createTag.tag) },
+      });
+    },
+  });
 
-  const [name, setName] = useState(tag ? tag.label : '');
-  const [description, setDescription] = useState(tag ? tag.description : '');
-  const [isActive, setIsActive] = useState(tag ? tag.is_active : false);
-  const [isReserved, setIsReserved] = useState(tag ? tag.is_reserved : false);
-  const [languageId, setLanguageId] = useState(tag ? tag.language_id : '');
-  const [parentId, setParentId] = useState(tag ? tag.parent_id : '');
-  const [formSubmitted, setFormSubmitted] = useState(false);
+  let tag: any = null;
+
+  useEffect(() => {
+    if (tagId && data) {
+      tag = tagId ? data.tag.tag : null;
+      setLabel(tag.label);
+      setDescription(tag.description);
+      setIsActive(tag.isActive);
+      setIsReserved(tag.isReserved);
+      setLanguageId(tag.language.id);
+      setParentId(tag.parent_id);
+    }
+  }, [data]);
+
+  if (loading) return <p>Loading...</p>;
+  if (error) return <p>Error :(</p>;
 
   const saveHandler = () => {
-    const payload: tagTypes.Tag = {
-      id: Number(tag ? tagId : Math.floor(Math.random() * Math.floor(100))),
-      label: name,
+    const payload = {
+      label: label,
       description: description,
-      is_active: isActive,
-      is_reserved: isReserved,
-      language_id: Number(languageId),
-      parent_id: Number(parentId),
+      isActive: isActive,
```

Comment:
Any specific reason for changing the keys and removing the type definition for this?

Reply:
Yeah. Couple of reasons:
1. The GraphQL requires camelCased keys in the payload.
2. in the type definition, we had id, which we don't need to pass to the API. It'll be set automatically.

Reply:
I think it's a good practice to keep types. We can do the following:
* Update types to reflect these changes.
* Remove id

Reply:
Makes sense. Working on it.

Reply:
This is complete. Ready for review again.